solutions/platinum/cses-2429.mdx

```diff
@@ -0,0 +1,128 @@
+---
+id: cses-2429
+source: CSES
+title: Grid Completion
+author: Alexis Tao
+---
+
+## Explanation
+
+This problem is all about finding the number of valid ways to configure a grid with some special conditions. The key is to break down the grid into manageable pieces.
+
+Save the effort of recalculating factorials and their modular inverses for the binomial coefficient by precomputing them using Fermat's Little Theorem. Then, track the character positions and count the rows and columns with A or B. Additionaly, define a helper function that returns the number of valid configurations based on how many rows we decide to fill with A’s and B’s. This function utilizes the inclusion-exclusion principle to adjust the counts. The outer loop iterates over all possible configurations and sums up the valid configurations.
```

Comment:
it would be good if you talked more about the intuition of arriving at the solution